variables.tf

```diff
@@ -1,9 +1,444 @@
+    error_message = "The specified archive_type is not a valid selection!"
+  }
+}
+
+variable "expire_days" {
+  description = "Specifies the number of days when the expire rule action takes effect. Only used if 'create_cos_bucket' is true."
+  type        = number
+  default     = null
+}
+
+variable "activity_tracker_crn" {
+  type        = string
+  description = "Activity tracker crn for COS bucket (Optional)"
+  default     = null
+  validation {
+    condition     = var.activity_tracker_crn != null ? can(regex("crn:v1:bluemix:public:logdnaat:(au-syd|eu-gb|eu-es|us-south|eu-de|us-east):a/[0-9a-f]{32}:[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}::", var.activity_tracker_crn)) : true
+    error_message = "Must be a valid activity tracker crn"
+  }
+}
+
+variable "sysdig_crn" {
+  type        = string
+  description = "Cloud Monitoring crn for COS bucket (Optional)"
+  default     = null
+}
+
+##############################################################################
+# COS bucket encryption variables
+##############################################################################
+
+variable "existing_kms_instance_guid" {
+  description = "The GUID of the Key Protect or Hyper Protect instance in which the key specified in var.kms_key_crn is coming from. Required if var.skip_iam_authorization_policy is false in order to create an IAM Access Policy to allow Key Protect or Hyper Protect to access the newly created COS instance."
+  type        = string
+  default     = null
+
+  validation {
+    condition     = var.existing_kms_instance_guid != null ? can(regex("^[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}$", var.existing_kms_instance_guid)) : true
+    error_message = "must be a valid cos instance id"
+  }
+}
+
+# variable "kms_encryption_enabled" {
+#   description = "Set as true to use KMS key encryption to encrypt data in COS bucket (only applicable when var.create_cos_bucket is true)."
+#   type        = bool
+#   default     = true
+# }
+
+# variable "kms_key_crn" {
+#   description = "CRN of the KMS key to use to encrypt the data in the COS bucket. Required if var.encryption_enabled and var.create_cos_bucket are true."
+#   type        = string
+#   default     = null
+# }
+
+##############################################################
+# Context-based restriction (CBR)
+##############################################################
+
+variable "bucket_cbr_rules" {
+  type = list(object({
+    description = string
+    account_id  = string
+    rule_contexts = list(object({
+      attributes = optional(list(object({
+        name  = string
+        value = string
+    }))) }))
+    enforcement_mode = string
+    tags = optional(list(object({
+      name  = string
+      value = string
+    })), [])
+    operations = optional(list(object({
+      api_types = list(object({
+        api_type_id = string
+      }))
+    })))
+  }))
+  description = "(Optional, list) List of CBR rules to create for the bucket"
```

Comment:
Exposing such a complex input variable means a consumer in projects will never be able to figure out the format required here without some supporting docs. The approach we have used is to create a helper markdown doc for complex inputs, and update the variable description to point to it. For example: https://github.com/terraform-ibm-modules/terraform-ibm-kms-all-inclusive/blob/main/solutions/standard/DA-keys.md